src/custom/state/operator/actions.ts

```diff
@@ -0,0 +1,18 @@
+import { createAction } from '@reduxjs/toolkit'
+
+// modeled after state/user/actions
+export interface SerializedToken {
+  chainId: number
+  address: string
+  decimals: number
+  symbol?: string
+  name?: string
+}
+
+export interface Tip {
+  value: string
```

Comment:
Probably we will need also a "minimal trade amount" @josojo can confirm if we can return a `expireDate` with the tip, so the operator commits to receive orders with

Reply:
Yeah, to my understanding, the best solution for the FE would be to check whether (sellAmount-tip.value)*price [USD] > "minimal trade amount" and warn the user if this is the gas. The idea is to warn users for spending to much gas on the tip. Though I am not sure whether you want the min tradeable amount in the tip interface.

Yeah, I think an `expireDate` makes sense but is not a must. My opinion is[ here](https://github.com/gnosis/oba-services/pull/49#discussion_r530988856). Currently, it is in the spec and I don't intend to kick it out :)

Reply:
> Yeah, I think an expireDate makes sense but is not a must. My opinion is here. Currently, it is in the spec and I don't intend to kick it out :)

Agree is not prio, but I think we should try to define the interfaces and given the effort it is, the backend could just return for example 30min expiring date, even if they don't implement it. This way, we can work with that. We usually underestimate UX, so is nice to just have it in our mind (to prevent user from getting errors when they send the tx)

> FE would be to check whether (sellAmount-tip.value)*price [USD] > "minimal trade amount" and warn the user if this is the gas.

I'm not sure we have the USD, also I don't think will be simple interface for other either. I also pictured this a bit different

We would allow to trade if:
* `sell amount  - tip > 0`

Then, we could also have the agreement with the backend that the formula for deciding how much tip we should pay would be:
* `ACTUAL_FEE = PAYED_FEE - MINIMAL_FEE `
* `ACTUAL_FEE = sell amount * 0.01% - MINIMAL_FEE`

@W3stside what do you think?

Reply:
> sell amount - tip > 0

yes makes sense, I've implemented a draft PR of the price estimation intercept which subtracts an arbitrary mock fee of '5' and if the amount sold is <=0 Uni blocks the trade anyways.

> ACTUAL_FEE = PAYED_FEE - MINIMAL_FEE
ACTUAL_FEE = sell amount * 0.01% - MINIMAL_FEE

Where is this information going to be used? Will this solely be for providers when they're setting the tip?

Reply:
Well, i called `ACTUAL_FEE` but it's just the tip. Used for calculating the min tokens you receive (sell orders) or the tokens max token you sell (buy order)

The quoting of the fee, for me just return the maximum fee for that token. that amount is what an operator needs in order to send the tx. But if we already pay some tokens with the fee (`sell amount * 0.01%`) we need to deduct that.